pygui/__init__.py

```diff
@@ -209,49 +214,67 @@ def reconcile_children(self, wip_fiber: Fiber, elements: List[VNode]):
             same_type = old_fiber and element and element.type == old_fiber.type
 
             if same_type:
-                # update the node
-                new_fiber = Fiber(
-                    type=old_fiber.type,
-                    props=element.props,
-                    # Have to create a snapshot of the state somewhere to
-                    # be able to reconcile properly... If the state has been
-                    # changed directly, there is no way of knowing what the
-                    # previous state looked like...
-                    # Another option is to let the renderer handle if there
-                    # should be any changes made to the dom object in the
-                    # `set_attribute` method :/ and thus leave out the comparison
-                    # between old and new props. That would also require to
-                    # combine the `set_attribute` and `clear_attribute`...
-                    # Vue's render API only has `patchProp`, but it still takes
-                    # a `prevValue` and a `nextValue`...
-                    raw_props=to_raw(element.props),
-                    children=element.children,
-                    dom=old_fiber.dom,
-                    parent=wip_fiber,
-                    alternate=old_fiber,
-                    effect_tag=EffectTag.UPDATE,
-                )
+                # Try and see if alternate of old_fiber is available for re-use
+                new_fiber = old_fiber.alternate or Fiber()
+                new_fiber.type = element.type
+                new_fiber.props = element.props
+                # Have to create a snapshot of the state somewhere to
+                # be able to reconcile properly... If the state has been
+                # changed directly, there is no way of knowing what the
+                # previous state looked like...
+                # Another option is to let the renderer handle if there
+                # should be any changes made to the dom object in the
+                # `set_attribute` method :/ and thus leave out the comparison
+                # between old and new props. That would also require to
+                # combine the `set_attribute` and `clear_attribute`...
+                # Vue's render API only has `patchProp`, but it still takes
+                # a `prevValue` and a `nextValue`...
```

Comment:
FWIW I think you are right to keep a snapshot of the previous props around (it can be shallow perhaps), and to compare them to the current prop values